File: typescript/tests/unit/context-file-urls.unit.test.ts
import { AgentBay } from "../../src";
import { ContextService } from "../../src/context";

function makeAgentBayWithMocks() {
  const agentBay = new AgentBay({ apiKey: "test-api-key" });
  const clientMock = (agentBay as any).getClient ? (agentBay as any).getClient() : (agentBay as any).client;
  return { agentBay, context: (agentBay as any).context as ContextService, clientMock };
}

describe("Context File URLs - Unit", () => {
  const contextId = "ctx-123";
  const testPath = "/tmp/integration_upload_test.txt";

  test("getFileUploadUrl success", async () => {
    const { context, clientMock } = makeAgentBayWithMocks();
    clientMock.getContextFileUploadUrl = jest.fn().mockResolvedValue({
      body: {
        success: true,
        data: { url: "https://oss.example.com/upload-url", expireTime: 3600 },
        requestId: "req-upload-1",
      },
    });

    const res = await context.getFileUploadUrl(contextId, testPath);
    expect(res.success).toBe(true);
    expect(res.url).toBe("https://oss.example.com/upload-url");
    expect(res.expireTime).toBe(3600);
  });

  test("getFileDownloadUrl success", async () => {
    const { context, clientMock } = makeAgentBayWithMocks();
    clientMock.getContextFileDownloadUrl = jest.fn().mockResolvedValue({
      body: {
        success: true,
        data: { url: "https://oss.example.com/download-url", expireTime: 7200 },
        requestId: "req-download-1",
      },
    });

    const res = await context.getFileDownloadUrl(contextId, testPath);
    expect(res.success).toBe(true);
    expect(res.url).toBe("https://oss.example.com/download-url");
    expect(res.expireTime).toBe(7200);
  });

  test("getFileDownloadUrl unavailable", async () => {
    const { context, clientMock } = makeAgentBayWithMocks();
    clientMock.getContextFileDownloadUrl = jest.fn().mockResolvedValue({
      body: {
        success: false,
        data: null,
        requestId: "req-download-2",
      },
    });

    const res = await context.getFileDownloadUrl(contextId, testPath);
    expect(res.success).toBe(false);
    expect(res.url).toBe("");
    expect(res.expireTime).toBeUndefined();
  });

  test("listFiles with entry", async () => {
    const { context, clientMock } = makeAgentBayWithMocks();
    clientMock.describeContextFiles = jest.fn().mockResolvedValue({
      body: {
        success: true,
        count: 1,
        data: [
          {
            fileId: "fid-1",
            fileName: "integration_upload_test.txt",
            filePath: "/tmp/integration_upload_test.txt",
            size: 21,
            status: "ready",
          },
        ],
        requestId: "req-list-1",
      },
    });

    const res = await context.listFiles(contextId, "/tmp", 1, 50);
    expect(res.success).toBe(true);
    expect(res.entries.length).toBe(1);
    expect(res.entries[0].filePath).toBe("/tmp/integration_upload_test.txt");
    expect(res.count).toBe(1);
  });

  test("listFiles empty", async () => {
    const { context, clientMock } = makeAgentBayWithMocks();
    clientMock.describeContextFiles = jest.fn().mockResolvedValue({
      body: {
        success: true,
        count: undefined,
        data: [],
        requestId: "req-list-2",
      },
    });

    const res = await context.listFiles(contextId, "/tmp", 1, 50);
    expect(res.success).toBe(true);
    expect(res.entries.length).toBe(0);
    expect(res.count).toBeUndefined();
  });

  test("deleteFile success", async () => {
    const { context, clientMock } = makeAgentBayWithMocks();
    clientMock.deleteContextFile = jest.fn().mockResolvedValue({
      body: {
        success: true,
        requestId: "req-del-1",
      },
    });

    const res = await context.deleteFile(contextId, testPath);
    expect(res.success).toBe(true);
  });
}); 

File: golang/tests/pkg/unit/context_file_urls_unit_test.go
package agentbay_test

import (
	"testing"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
	interfaces "github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay/interface"
	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay/models"
	"github.com/stretchr/testify/assert"
)

// fakeContext is a simple fake implementing interfaces.ContextInterface for unit tests
type fakeContext struct {
	uploadURLRes   *agentbay.ContextFileUrlResult
	downloadURLRes *agentbay.ContextFileUrlResult
	listFilesRes   *agentbay.ContextFileListResult
	deleteFileRes  *agentbay.ContextFileDeleteResult
}

// File URL APIs
func (f *fakeContext) GetFileUploadUrl(contextID string, filePath string) (*agentbay.ContextFileUrlResult, error) {
	return f.uploadURLRes, nil
}

func (f *fakeContext) GetFileDownloadUrl(contextID string, filePath string) (*agentbay.ContextFileUrlResult, error) {
	return f.downloadURLRes, nil
}

func (f *fakeContext) ListFiles(contextID string, parentFolderPath string, pageNumber int32, pageSize int32) (*agentbay.ContextFileListResult, error) {
	return f.listFilesRes, nil
}

func (f *fakeContext) DeleteFile(contextID string, filePath string) (*agentbay.ContextFileDeleteResult, error) {
	return f.deleteFileRes, nil
}

// Other methods to satisfy interfaces.ContextInterface (not used in these tests)
func (f *fakeContext) List(params *agentbay.ContextListParams) (*agentbay.ContextListResult, error) {
	return nil, nil
}
func (f *fakeContext) Get(name string, create bool) (*agentbay.ContextResult, error) { return nil, nil }
func (f *fakeContext) Create(name string) (*agentbay.ContextCreateResult, error)     { return nil, nil }
func (f *fakeContext) Update(context *agentbay.Context) (*agentbay.ContextModifyResult, error) {
	return nil, nil
}
func (f *fakeContext) Delete(context *agentbay.Context) (*agentbay.ContextDeleteResult, error) {
	return nil, nil
}

// Helper to get a ContextInterface from fake
func newFakeContext(fc *fakeContext) interfaces.ContextInterface { return fc }

func TestContext_GetFileUploadUrl_Unit(t *testing.T) {
	// Arrange
	expire := int64(3600)
	fake := newFakeContext(&fakeContext{
		uploadURLRes: &agentbay.ContextFileUrlResult{
			ApiResponse: models.WithRequestID("req-upload-1"),
			Success:     true,
			Url:         "https://oss.example.com/upload-url",
			ExpireTime:  &expire,
		},
	})

	// Act
	res, err := fake.GetFileUploadUrl("ctx-123", "/tmp/integration_upload_test.txt")

	// Assert
	assert.NoError(t, err)
	assert.NotNil(t, res)
	assert.True(t, res.Success)
	assert.Equal(t, "https://oss.example.com/upload-url", res.Url)
	if assert.NotNil(t, res.ExpireTime) {
		assert.Equal(t, int64(3600), *res.ExpireTime)
	}
}

func TestContext_GetFileDownloadUrl_Unit(t *testing.T) {
	// Arrange
	expire := int64(7200)
	fake := newFakeContext(&fakeContext{
		downloadURLRes: &agentbay.ContextFileUrlResult{
			ApiResponse: models.WithRequestID("req-download-1"),
			Success:     true,
			Url:         "https://oss.example.com/download-url",
			ExpireTime:  &expire,
		},
	})

	// Act
	res, err := fake.GetFileDownloadUrl("ctx-123", "/tmp/integration_upload_test.txt")

	// Assert
	assert.NoError(t, err)
	assert.NotNil(t, res)
	assert.True(t, res.Success)
	assert.Equal(t, "https://oss.example.com/download-url", res.Url)
	if assert.NotNil(t, res.ExpireTime) {
		assert.Equal(t, int64(7200), *res.ExpireTime)
	}
}

func TestContext_GetFileDownloadUrl_Unit_Unavailable(t *testing.T) {
	// Arrange
	fake := newFakeContext(&fakeContext{
		downloadURLRes: &agentbay.ContextFileUrlResult{
			ApiResponse: models.WithRequestID("req-download-2"),
			Success:     false,
			Url:         "",
			ExpireTime:  nil,
		},
	})

	// Act
	res, err := fake.GetFileDownloadUrl("ctx-123", "/tmp/integration_upload_test.txt")

	// Assert
	assert.NoError(t, err)
	assert.NotNil(t, res)
	assert.False(t, res.Success)
	assert.Equal(t, "", res.Url)
	assert.Nil(t, res.ExpireTime)
}

func TestContext_ListFiles_Unit_WithEntry(t *testing.T) {
	// Arrange
	entries := []*agentbay.ContextFileEntry{
		{
			FileID:   "fid-1",
			FileName: "integration_upload_test.txt",
			FilePath: "/tmp/integration_upload_test.txt",
			Size:     21,
			Status:   "ready",
		},
	}
	count := int32(1)
	fake := newFakeContext(&fakeContext{
		listFilesRes: &agentbay.ContextFileListResult{
			ApiResponse: models.WithRequestID("req-list-1"),
			Success:     true,
			Entries:     entries,
			Count:       &count,
		},
	})

	// Act
	res, err := fake.ListFiles("ctx-123", "/tmp", 1, 50)

	// Assert
	assert.NoError(t, err)
	assert.NotNil(t, res)
	assert.True(t, res.Success)
	assert.Equal(t, 1, len(res.Entries))
	assert.Equal(t, "/tmp/integration_upload_test.txt", res.Entries[0].FilePath)
	if assert.NotNil(t, res.Count) {
		assert.Equal(t, int32(1), *res.Count)
	}
}

func TestContext_ListFiles_Unit_Empty(t *testing.T) {
	// Arrange
	fake := newFakeContext(&fakeContext{
		listFilesRes: &agentbay.ContextFileListResult{
			ApiResponse: models.WithRequestID("req-list-2"),
			Success:     true,
			Entries:     []*agentbay.ContextFileEntry{},
			Count:       nil,
		},
	})

	// Act
	res, err := fake.ListFiles("ctx-123", "/tmp", 1, 50)

	// Assert
	assert.NoError(t, err)
	assert.NotNil(t, res)
	assert.True(t, res.Success)
	assert.Equal(t, 0, len(res.Entries))
	assert.Nil(t, res.Count)
}

func TestContext_DeleteFile_Unit(t *testing.T) {
	// Arrange
	fake := newFakeContext(&fakeContext{
		deleteFileRes: &agentbay.ContextFileDeleteResult{
			ApiResponse: models.WithRequestID("req-del-1"),
			Success:     true,
		},
	})

	// Act
	res, err := fake.DeleteFile("ctx-123", "/tmp/integration_upload_test.txt")

	// Assert
	assert.NoError(t, err)
	assert.NotNil(t, res)
	assert.True(t, res.Success)
}
